esmvaltool/recipes/ref/recipe_seaborn_trend_regions.yml

```diff
@@ -0,0 +1,309 @@
+# ESMValTool
+---
+documentation:
+  description:
+    Box plot, linear trends for
+    the 46 IPCC AR6 reference regions for land (Iturbide et al., 2020).
+    The diagnostic called in this recipe uses the seaborn.boxplot function.
+    See seaborn manual for detailed information
+    (https://seaborn.pydata.org/generated/seaborn.boxplot.html).
+
+  authors:
+    - schlund_manuel
+    - weigel_katja
+  maintainer:
+    - weigel_katja
+  title: Linear trends.
+
+datasets:
+  - {dataset: MIROC6,      project: CMIP6, exp: historical, ensemble: r1i1p1f1, grid: gn}
+
+preprocessors:
+  trend_tas: &trend
+    custom_order: true
+    extract_shape:
+      shapefile: ar6
+      decomposed: true
+      ids:
+        Acronym:
+          - 'NWN'
+          - 'NEN'
+          - 'GIC'
+          - 'WNA'
+          - 'CNA'
+          - 'ENA'
+          - 'NCA'
+          - 'SCA'
+          - 'CAR'
+          - 'NWS'
+          - 'NSA'
+          - 'NES'
+          - 'SAM'
+          - 'SWS'
+          - 'SES'
+          - 'SSA'
+          - 'NEU'
+          - 'WCE'
+          - 'EEU'
+          - 'MED'
+          - 'SAH'
+          - 'WAF'
+          - 'CAF'
+          - 'NEAF'
+          - 'SEAF'
+          - 'WSAF'
+          - 'ESAF'
+          - 'MDG'
+          - 'RAR'
+          - 'WSB'
+          - 'ESB'
+          - 'RFE'
+          - 'WCA'
+          - 'ECA'
+          - 'TIB'
+          - 'EAS'
+          - 'ARP'
+          - 'SAS'
+          - 'SEA'
+          - 'NAU'
+          - 'CAU'
+          - 'EAU'
+          - 'SAU'
+          - 'NZ'
+          - 'WAN'
+          - 'EAN'
+    area_statistics:
+      operator: mean
+    linear_trend:
+    convert_units:
+      units: K yr-1
+
+  trend_pr:
+    <<: *trend
+    convert_units:
+      units: mm kg m-3 day-1 yr-1
+
+  trend_psl:
+    <<: *trend
+    convert_units:
+      units: m-1 kg s-2 yr-1
+
+
+  trend_ua200: &trend200
+    custom_order: true
+    extract_levels:
+      levels: 20000
+      scheme: linear
+      coordinate: air_pressure
+    extract_shape:
+      shapefile: ar6
+      decomposed: true
+      ids:
+        Acronym:

```

Comment:
I think it would be better to have the trends available for each individual region, not just as a collection of all regions. Most likely the results will be assembled in a new way (e.g. time series, annual cycle and trends per model per region) from the REF anyway, and then it might be better to just have the trend bar plot available for each individual region.

Reply:
I'm not sure if that is really the case, but for consistency I can do it for single regions. Unfortunately, there is an iris bug causing issues with pandas (as @schlunma mentioned above), so we cannot simply plot only one region in the same style as many at the moment. But I managed to find a recipe setting, which allows it in anyway (using dataset both as x and hue). With this, the plots would look like this:
![seaborn_barplot](https://github.com/user-attachments/assets/fcd8b004-a308-4354-bee1-7fecd4b967ae)

Would you prefer this version?

Reply:
That would be great! I know for sure that the REF will be doing all kinds of additional grouping and presentation of the results. So having the trends in for one model and one region in one bar plot would be great!

Reply:
Unfortunately it seems not to be possible. Only for the 3D variables I managed to set one region but different air-pressure levels:
![seaborn_barplot](https://github.com/user-attachments/assets/8b540b12-62b4-4446-9abe-d73b78be297d)

(but at leas for hus that is maybe not the best choice because of its huge vertical gradient)
For the 2D ones I did not find any other split within a region, which works so far (I'd hoped for zonal_statistics, but it has an issue with NaNs)

Reply:
@hb326 if we really need one model + one variable + one region now, I guess I could try to switch to a different diagnostic, there are some more which do bar plots (e.g. one of the deangelis ones). But I doubt that I would manage without changing a diagnositc.

Reply:
Thanks @hb326 ! So I left the trend recipe now as it is and changed the other ones following your comments.

Reply:
I made an error with double default, update soon

Reply:
fixed :)